python/action-server/state_machine.py

```diff
@@ -0,0 +1,78 @@
+from transitions.extensions import GraphMachine
+from transitions.extensions.markup import MarkupMachine
+from transitions.extensions.factory import HierarchicalMachine
+from transitions import Machine
+import logging
+import json
+import time
+from IPython.display import Image, display, display_png
+
+log = logging.getLogger(__name__)
+
+QUEUED = True
+
+class Session_state:
+    def init_session(self, session):
+        self.session=session
+
+    def statechart(self):
+        statechart = json.dumps(self.markup, indent=3)
+        self.session.put("/statechart", statechart)
+
+    def state_(self):
+        self.session.put("/state", self.state)
+        
+
+    def trigger(self, trigger_method):
+        try:
+            self.session.put("/trigger/"+trigger_method, {'timestamp':time.time(), 'triggered':trigger_method})
+        except Exception as e:
+            self.session.put("/error/"+trigger_method, {'timestamp':time.time(), 'Error': e})
+
+
+class Unhealthy(HierarchicalMachine, Session_state):
+    def __init__(self):
+        states = [{"name":'aborted', "on_enter":["aborted"]},
```

Comment:
inlining the sm makes more sense rather than having it in an external json, so this looks better